scripts/install-chef-suse.sh

```diff
@@ -665,7 +665,9 @@ echo_summary "Performing initial chef-client run"
 service chef-client status &> /dev/null && service chef-client stop
 
 if ! [ -e ~/.chef/knife.rb -a -e ~/.chef/root.pem ]; then
-    yes '' | knife configure -i
+    # no_proxy is currently not suppoted in ruby see bsc#958716
+    # unset it for this call
+    (unset http_proxy; yes '' | knife configure -i)
```

Comment:
install-chef-suse.sh runs with `set -e`
Moving the knife call to a subshell would prevent the -e to exit install-chef-suse.sh if knife exits with an error
So $? needs to be evaluated right after the subshell and exit if it is != 0